### ISSUE TITLE
`name` is not reserved keyword in Postgres

### ISSUE THREAD

**Issue by hongbo-miao** [sqlfluff/sqlfluff#7233]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

When upgrade sqlfluff from `3.4.2` to `3.5.0`, I got errors such as

```sh
== [up.sql] FAIL                                                                                                                                                                                                      
L:  13 | P:   5 | RF04 | Keywords should not be used as identifiers.                                                                                                                                                                                                                                     
                       | [references.keywords]
```

for line


```sql
create or replace function trigger_update_updated_at()
returns trigger as
$$
begin
  new.updated_at = now();
  return new;
end;
$$ language plpgsql;

create table if not exists opal_client
(
    id uuid default gen_random_uuid(),
    name text not null,  -- This line caused the issue
    config text,
    created_at timestamptz not null default now(),
    updated_at timestamptz not null default now(),
    primary key (id)
);
```

However, `name` is not reserved keyword in Postgres based on

https://www.postgresql.org/docs/current/sql-keywords-appendix.html

<img width="2005" height="266" alt="Image" src="https://github.com/user-attachments/assets/db0fda35-369e-4e43-acc2-144278ad194a" />

### Expected Behaviour

Expect no errors.

### Observed Behaviour

Throw errors.

### How to reproduce

Run

```sh
sqlfluff lint --dialect=postgres folder
```

with file `up.sql` inside.

```sql
create or replace function trigger_update_updated_at()
returns trigger as
$$
begin
  new.updated_at = now();
  return new;
end;
$$ language plpgsql;

create table if not exists opal_client
(
    id uuid default gen_random_uuid(),
    name text not null,  -- This line caused the issue
    config text,
    created_at timestamptz not null default now(),
    updated_at timestamptz not null default now(),
    primary key (id)
);
```

### Dialect

postgres

### Version

```sh
sqlfluff, version 3.5.0
```

### Configuration

.sqlfluff
```
[sqlfluff]
exclude_rules = layout.long_lines
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by ChristianBeilschmidt** [comment]
This is also true for at least:
```
    class,
    data,
    depth,
    family,
    hash,
    location,
    name,
    summary
```

**Comment by WittierDinosaur** [comment]
This is expected behaviour, reserved keywords are not parseable, as they cannot be identifiers. RF04 is that non-reserved keywords should not be used as identifiers. If that's not a rule you agree with, turn it off in your config

**Comment by hongbo-miao** [comment]
Hi @WittierDinosaur, I agree on

> RF04 is that non-reserved keywords should not be used as identifiers.

The key question here is whether “name” is actually reserved. I showed that the official PostgreSQL documentation lists “name” as not reserved. Since you mentioned it is reserved, could you please share the reference link? Do you mean official Postgres document is wrong?

**Comment by hongbo-miao** [comment]
Hi @WittierDinosaur do you mind reopening this? Thanks

**Comment by hongbo-miao** [comment]
Recreated at https://github.com/sqlfluff/sqlfluff/issues/7269
